File: Service/Z2data.Core/Domain/Locations/Seaport.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Z2data.Core.Domain.Events;
using Z2data.Core.Domain.Lookups;

namespace Z2data.Core.Domain.Locations
{
    public class Seaport : BaseEntity
    {
        public string Name { get; set; }
        public Lookup Type { get; set; }
        public Place Place { get; set; }
        public Location Location { get; set; }
        public string LoCode { get; set; }
    }
}

File: Front-end/src/app/locations/seaport.ts
import { BaseEntity } from '../shared/baseEntity';
import { Lookup } from '../lookups/lookup';
import { Country } from '../lookups/country';
import { City } from '../lookups/city';

export interface Seaport extends BaseEntity {
    /** public int PortId { get; set; } */
    name: string;
    type: Lookup;
    place: {
        country: Country;
        state: {
            stateId: number;
            name: string;
        };
        city: City;
        other: string;
    };
    location: Location;
    loCode: string;
}
